### PR TITLE
Use try instead of unwrap in doctests

### PR DESCRIPTION
**Pull request by spacejam**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -509,12 +509,11 @@ impl<TypeState> ConnectionOptions<TypeState> {
     ///
     /// # Examples
     /// ```no_run
-    /// # fn main() -> std::io::Result<()> {
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
```
Comment (spacejam) [inline src/lib.rs:512]: this allows most errors to be converted into the return type using the try `?` operator

File: src/lib.rs [hunk 2]
```diff
@@ -487,7 +487,7 @@ impl<TypeState> ConnectionOptions<TypeState> {
     /// to apply the desired configuration to all server connections.
     ///
     /// # Examples
-    /// ```
+    /// ```no_run
```
Comment (derekcollison) [inline src/lib.rs:490]: Why no run?

Comment (spacejam) [reply src/lib.rs:490]: sometimes the test fails from my home connection to the demo server. I'll run an mtr next time and find out why.

Comment (derekcollison) [reply src/lib.rs:490]: We must have a timeout of some sort too low. Since we are going Germany to west coast US. Let's fix that and keep running it imo.

Comment (spacejam) [reply src/lib.rs:490]: Loss sometimes generally just gets kind of high depending on the route. Today's weather event seems to be around the Telia backbone. Often there are weather events related to poor network maintainership of German ISPs. So, in this case it seems to be more of a normal internet weather thing. It errors out fairly frequently and tends to be a source of false-negative stress.
image

Comment (derekcollison) [reply src/lib.rs:490]: How long does the client try to connect currently?

**Pull request merged**